examples/systemcalls/systemcalls.c

```diff
@@ -92,8 +194,77 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    //open the file to redirect STDOUT to
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+    if(fd<0)
+    {
+        //if open failed
+        perror("Open");
+        return false;
+    }
 
-    va_end(args);
-
-    return true;
+    
+    pid_t pid;
+    pid = fork ();
+    if (pid == -1)
+    {
+        //if fork failed
+        perror("Fork");
+        va_end(args);
+        return false;
+    }
+    //if child process
+    else if(pid == 0)
+    {
+        //fd and 1 (which is STDOUT) both point to the same thing now i.e, STDOUT points to fd
+        if (dup2(fd, 1) < 0)
+        {
+            //if dup failed. 
+            perror("dup2"); 
+            return false; 
+        }
+        close(fd);
+        //exec the command
+        if(execv (command[0], (command))== -1)
+        {            
+            //if exec returned
+            perror("Exec");
+            va_end(args);
+            exit(-1);
+        }
+    }
+    //if parent process
+    else
+    {
+        close(fd);
+        //wait for the child
+        if (waitpid (pid, &status, 0) == -1)
+        {
+            perror("Wait");
+            va_end(args);
+            return false;
+        }
+        //if child exited normally
+        else if (WIFEXITED (status))
+        {
+            int ret_status = WEXITSTATUS(status);
+            //check exit status of child
+            if(ret_status!=0)
+            {
+                printf("Command exited with non-zero status %d\n",ret_status);
+                va_end(args);
+                return false;
+            }
+            //if child returned normally and exit status = 0, then return true
+            else
+            {
+                va_end(args);
+                return true;
+            }
+        }        
+        //if child did not return normally return false
+        va_end(args);
+        return false;
```

Comment:
@GuruprashanthKrishnakumar think about how you could avoid duplicating logic between these two `do_exec` functions